libs/base/ui/structure/text/text.component.ts

```diff
@@ -11,6 +11,76 @@ export class TextComponent extends LitElement implements TextProperties {
 
   protected render(): TemplateResult | void {
     if (!this.content) return;
-    return html`${unsafeHTML(this.content)}`;
+    const c = this.appendStyleToHeaders(this.content);
+    return html`${unsafeHTML(c)}`;
+  }
+
+  /**
+   * Modifies an HTML string by adding custom inline styles to specific HTML elements
+   * based on their tags, classes, and screen sizes. The styles align roughly with the
+   * Oryx design system without reusing the design system components directly.
+   *
+   * @param htmlString - The HTML string to be modified.
+   * @returns The modified HTML string with custom inline styles.
+   */
+  protected appendStyleToHeaders(htmlString: string): string {
```

Comment:
Maybe better to extract this logic to the controller? It will be much more flexible, If we need this logic in some another component, just add the controller instead of `<oryx-text>` inside render, in `oryx-content-text` for example

Reply:
We've tried a lot of variations. We ended up with a reusable text mixin, but it's only concerned with converting html to Oryx design system and no longer comes with opinions on the styling of the text, since this is part of the design system.